Harvest.Web/ClientApp/src/Requests/AccountChangeContainer.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+
+import { Project, ProjectAccount } from "../types";
+import { AccountsInput } from "./AccountsInput";
+import { ProjectHeader } from "./ProjectHeader";
+
+interface RouteParams {
+  projectId?: string;
+}
+
+export const AccountChangeContainer = () => {
+  const { projectId } = useParams<RouteParams>();
+  const [project, setProject] = useState<Project | undefined>();
+  const [accounts, setAccounts] = useState<ProjectAccount[]>([]);
+  const [disabled, setDisabled] = useState<boolean>(true);
+
+  useEffect(() => {
+    const cb = async () => {
+      const response = await fetch(`/Project/Get/${projectId}`);
+
+      if (response.ok) {
+        const proj: Project = await response.json();
+
+        setProject(proj);
+      }
+    };
+
+    cb();
+  }, [projectId]);
+
+  if (project === undefined) {
+    return <div>Loading ...</div>;
+  }
+
+  //TODO: require PI or supervisor access after updating auth policies
+  const changeAccounts = async () => {
+    const response = await fetch(`/Request/Approve/${projectId}`, {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ Accounts: accounts }),
+    });
+
+    if (response.ok) {
+      const data = await response.json();
+      window.location.pathname = `/Project/Details/${data.id}`;
```

Comment:
@wongband  we shouldn't re-route this way since it forces a full-page refresh.  Would be better to `useHistory` and push the new state, or use the <Redirect> component of react-router https://reactrouter.com/web/api/Hooks/usehistory.

Reply:
If you want to do this in a separate PR we can.  There might be other places in the app doing this as well, and if so we should change them too.  I wrote a few like this from before those pages were react, but once all the pages became react we should then route doing client-side routing only.

Reply:
Okay got it. I will change that

Reply:
@srkirkland I fixed the routing in that component. I found a couple others as well. Do you want me to fix those in a separate PR?

Reply:
@wongband perfect I think this is ready to merge and then yes, please fix the others in a different PR